ow_regolith/include/DigStateMachine.h

```diff
@@ -0,0 +1,176 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+// Defines a finite state machine and associated states that estimate the phase
+// of a digging operation performed with the lander scoop.
+
+#ifndef DIG_FSM_H
+#define DIG_FSM_H
+
+#include <ros/ros.h>
+#include <tf/tf.h>
+
+#include <string>
+#include <memory>
+
+namespace ow_regolith {
+
+class DigStateMachine;
+
+class DigState
+{
+public:
+  DigState(const std::string &name, DigStateMachine *const c)
+    : m_name(name), m_context(c) { };
+  virtual ~DigState() = default;
+
+  DigState() = delete;
+  DigState(const DigState&) = delete;
+  DigState& operator=(const DigState&) = delete;
+
+  const std::string& getName() const {return m_name;}
+
+  // external events DigStates will handle or ignore
+  virtual void terrainModified() { };
+  virtual void scoopPoseUpdate() { };
+
+  virtual void enter() { };
+  virtual void exit() { };
+
+protected:
+  DigStateMachine *const m_context;
+private:
+  const std::string m_name;
+};
+
+class Timeout
```

Comment:
Since this class is small I suggest incorporating the timeout into DigState and eliminating the multiple inheritance (a complexity that's good to avoid in general).  Otherwise, put this inside DigStateMachine if it's not to be used elsewhere.

Reply:
I think I prefer this as a multiple inheritance because NotDiggingState does not require a timeout at all, so to put it in the base class would mean one of the states has a member variable that's never used. Timeout and DigState have both been made inner classes of DigStateMachine. Let me know if you think that's enough.

Reply:
Per suggestion I made in Dev Day, since this inheritance does not model a "is a kind of " relationship, aggregation is a better approach: DigState can contain a Timeout (e.g. as a pointer).  It's okay if not all the subclasses use it.

Reply:
Timeout is now removed and its features are now instead handled by a timeout member in the base class DigState. I was able to the timeout's instantiation only occur if the right DigState constructor is called by using `std::optional`.